addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -2770,6 +2770,12 @@ const SnippetOptionWidget = Widget.extend({
      * @type {boolean}
      */
     displayOverlayOptions: false,
+    /**
+     * The option should not be available for inline snippets
+     *
+     * @type {boolean}
+     */
+    disabledForInlineSnippets: false,
```

Comment:
I don't think it should be a new property in the option JS class but rather just an update of the `data-selector` of the related option XML (instead of `data-selector="section"`, it should be `data-selector="section" data-exclude=".row *"` to exclude those option to exist on snippets in columns for example). Maybe it is worth a shortcut like you did but still, in the XML as this is the same principle as the data-selector and data-exclude attributes (but I am not even sure, `data-exclude=".row *"` is easy to understand and does not add another attribute to document to the system)